src/dmd/typesem.d

```diff
@@ -3091,12 +3093,14 @@ void resolve(Type mt, const ref Loc loc, Scope* sc, out Expression pe, out Type
         {
             e = e.expressionSemantic(sc);
             if (auto et = e.isTypeExp())
-                return returnType(et.type.addMod(mt.mod));
+                returnType(et.type.addMod(mt.mod));
             else
                 returnExp(e);
         }
         else
             returnError();
+
+        mt.obj = pe ? pe : (pt ? pt : ps);
```

Comment:
Same thing here. Comment on why please.

Reply:
Here the opposite. I'm saving the semantic result.